lib/using_restricted_array.rb

```diff
@@ -1,61 +1,118 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear or (O)n, where n is the number of items in
+# the array or the length of the input array
+# Space complexity: constant O(1)
 def length(array)
-  raise NotImplementedError
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear O(n), where n is the lenght of the array
+# Space complexity: constant O(1)
+
 def print_array(array)
-  raise NotImplementedError
+  len = length(array)
+  i = 0
+  while i < len
+    puts array[i]
+    i += 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of values in the array
+# Space complexity: constant O(1)
 def search(array, length, value_to_find)
-  raise NotImplementedError
+  length.times do |i|
```

Comment:
[nit] It's fine to use length since it's given here, but I think it's more robust to use the same until array[i] == nil method. Someone could call your function with an incorrect length.